Installation issue: py-torch

### Steps to reproduce the issue

Hello, I have a build error with `py-torch` since [this commit](https://github.com/spack/spack/commit/bb64b22066d359ace40c944cc42ab5ff34bf8c7e). `py-torch:2.1.0` was building fine on my laptop before that commit.

```console
$ spack spec -I py-torch@2.1 ~gloo ~fbgemm ~kineto +distributed
```
@adamjstewart 

### Error message

```
/home/tbouvier/Dev/spack-envs/local/.spack-env/view/bin/ld: /home/tbouvier/Dev/spack/opt/spack/linux-fedora38-skylake/gcc-13.1.1/sleef-3.5.1_2020-12-22-jm5alvqtwrfed6lcx7ct4ekzw4ewptjs/lib64/libsleef.a(dispavx.c.o): relocation R_X86_64_32 against symbol `Sleef_fmad4_fma4' can not be used when making a shared object; recompile with -fPIC
```

<details><summary>Error message</summary>

<pre>
  [5366/5504] Linking CXX shared library lib/libtorch_cpu.so
  FAILED: lib/libtorch_cpu.so

  /home/tbouvier/Dev/spack-envs/local/.spack-env/view/bin/ld: /home/tbouvier/Dev/spack/opt/spack/linux-fedora38-skylake/gcc-13.1.1/sleef-3.5.1_2020-12-22-jm5alvqtwrfed6lcx7ct4ekzw4ewptjs/lib64/libsleef.a(dispavx.c.o): relocation R_X86_64_32 against symbol `Sleef_fmad4_fma4' can not be used when making a shared object; recompile with -fPIC
  /home/tbouvier/Dev/spack-envs/local/.spack-env/view/bin/ld: failed to set dynamic section sizes: bad value
  collect2: error: ld returned 1 exit status
</pre></details>


### Information on your system

* **Spack:** 0.21.0.dev0 (668a5b45e5ca5b5c16ea042d0121c5a958564089)
* **Python:** 3.11.6
* **Platform:** linux-fedora38-skylake
* **Concretizer:** clingo


### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/13256493/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/13256494/spack-build-out.txt)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/sleef/package.py
+++ b/var/spack/repos/builtin/packages/sleef/package.py
@@ -53,10 +53,5 @@ class Sleef(CMakePackage):
     # conflicts("^mpfr@4.2:")
 
     def cmake_args(self):
-        # Taken from PyTorch's aten/src/ATen/CMakeLists.txt
-        return [
-            self.define("BUILD_SHARED_LIBS", False),
-            self.define("BUILD_DFT", False),
-            self.define("BUILD_GNUABI_LIBS", False),
-            self.define("BUILD_TESTS", False),
-        ]
+        # https://salsa.debian.org/science-team/sleef/-/blob/master/debian/rules
+        return [self.define("BUILD_DFT", False), self.define("SLEEF_TEST_ALL_IUT", True)]

--- a/var/spack/repos/builtin/packages/sleef/package.py
+++ b/var/spack/repos/builtin/packages/sleef/package.py
@@ -54,4 +54,8 @@ class Sleef(CMakePackage):
 
     def cmake_args(self):
         # https://salsa.debian.org/science-team/sleef/-/blob/master/debian/rules
-        return [self.define("BUILD_DFT", False), self.define("SLEEF_TEST_ALL_IUT", True)]
+        return [
+            self.define("BUILD_DFT", False),
+            self.define("SLEEF_TEST_ALL_IUT", True),
+            self.define("BUILD_TESTS", False),
+        ]

